src/gallery/slider.js

```diff
@@ -0,0 +1,231 @@
+			loading.style.visibility = 'hidden'
+		} else {
+			const textElement = container.querySelector( `.${prefixClassname}-item-loading-text` ),
+				timeoutId = setTimeout( () => {
+					textElement.style.visibility = 'visible'
+				}, 5000 )
+
+			imageElement.addEventListener( 'load', () => {
+				loading.style.visibility = 'hidden'
+				textElement.style.visibility = 'hidden'
+				clearTimeout( timeoutId )
+			} )
+
+			imageElement.addEventListener( 'error', () => {
+				const refreshElement = container.querySelector( `.${prefixClassname}-item-loading-error-refresh` )
+				loading.style.visibility = 'hidden'
+				errorElement.style.visibility = 'visible'
+				clearTimeout( timeoutId )
+
+				refreshElement.addEventListener( 'click', () => {
+					bindImageEvent( container, true )
+				} )
+			} )
+
+		}
+	},
+
+	renderNext = ( offset = 1 ) => {
+		const slider = parentContainer.querySelector( `.${prefixClassname}` ),
+			items = slider.querySelectorAll( `.${prefixClassname}-item` ),
+			nextButton = slider.querySelector( '.next' ),
+			previousButton = slider.querySelector( '.previous' ),
+			next = current + offset,
+			item = items[ next ]
+
+		if ( item ) {
+			const caption = item.querySelector( `.${prefixClassname}-item-caption` ),
+				attribution = item.querySelector( `.${prefixClassname}-item-attribution` )
+
+			current += offset
+			nextButton.style.opacity = current === items.length - 1 ? '0.5' : '1'
+			previousButton.style.opacity = current === 0 ? '0.5' : '1'
+
+			bindImageEvent( item )
+
+			// render image attribution element
+			if ( !caption && !attribution ) {
+				requestPageMediaInfo(
+					lang,
+					gallery[ next ].title,
+					gallery[ next ].fromCommon,
+					mediaInfo => {
+						item.insertAdjacentHTML(
+							'beforeend',
+							renderImageInfo( mediaInfo, gallery[ next ], lang
+							) )
+					} )
+			}
+		}
+
+		slider.style[ dir === 'ltr' ? 'marginLeft' : 'marginRight' ] = -clientWidth * current + 'px'
+	},
+
+	renderPrevious = () => {
+		renderNext( -1 )
+	},
+
+	applyGestureEvent = () => {
+		let temp = {
+			screenX: null,
+			targetScreenX: null,
+			originalMarginLeft: null,
+			currentMarginLeft: null,
+			originalTransition: null
+		}
+
+		const container = parentContainer.querySelector( `.${prefixClassname}` ),
+			marginLR = dir === 'ltr' ? 'marginLeft' : 'marginRight'
+
+		container.addEventListener( 'touchstart', e => {
+			temp.screenX = e.touches[ 0 ].clientX
+			temp.targetScreenX = null
+			temp.originalMarginLeft =
+                +window.getComputedStyle( container )[ marginLR ].slice( 0, -2 )
+			temp.currentMarginLeft =
+                +window.getComputedStyle( container )[ marginLR ].slice( 0, -2 )
+			temp.originalTransition = window.getComputedStyle( container ).transition
+			container.style.transition = 'unset'
+		} )
+		container.addEventListener( 'touchmove', e => {
+			const clientX = e.touches[ 0 ].clientX,
+				offset = clientX - temp.screenX
+			temp.targetScreenX = clientX
```

Comment:
Does `temp.targetScreenX` have a specific purpose? Seems to be working fine without it

Reply:
true! will remove that!